launch.md

```diff
@@ -0,0 +1,150 @@
+ 
+**customLaunchSetupCommands**: If provided, this replaces the default commands used to launch a target with some other commands. For example, this can be "-target-attach" in order to attach to a target process. An empty command list replaces the launch commands with nothing, which can be useful if the debugger is being provided launch options as command line options. Example: `"customLaunchSetupCommands": [ { "text": "target-run", "description": "run target", "ignoreFailures": false }]`.
+      
+**launchCompleteCommand**: The command to execute after the debugger is fully setup in order to cause the target process to run. Allowed values are "exec-run", "exec-continue", "None". The default value is "exec-run".
+
+**Example:**
+
+```json
+{
+   "name": "C++ Launch",
+   "type": "cppdbg",
+   "request": "launch",
+   "program": "${workspaceRoot}/a.out",
+   "stopAtEntry": false,
+   "customLaunchSetupCommands": [
+      { "text": "target-run", "description": "run target", "ignoreFailures": false }
+   ],
+   "launchCompleteCommand": "exec-run",
+   "linux": {
+      "MIMode": "gdb",
+      "miDebuggerPath": "/usr/bin/gdb"
+   },
+   "osx": {
+      "MIMode": "lldb"
+   },
+   "windows": {
+      "MIMode": "gdb",
+      "miDebuggerPath": "C:\\MinGw\\bin\\gdb.exe"
+   }
+}
+```
+
+## Debugging dump files
+
+The C/C++ extension enables debugging dump files on Windows and core dump files Linux and OS X.
+
+**dumpPath**: If you want to debug a Windows dump file, set this to the path to the dump file to start debugging in the `launch` configuration.
+
+**coreDumpPath**: Full path to a core dump file to debug for the specified program.  Set this to the path to the core dump file to start debugging in the `launch` configureation. 
+
+## Remote debugging or debugging with a local debugger server
+
+**miDebuggerServerAddress**: Network address of the debugger server (e.g. gdbserver) to connect to for remote debugging (example: `localhost:1234`).
+
+**debugServerPath**: Full path to debug server to launch.
+
+**debugServerArgs**: Arguments for the the debugger server.
+                
+**serverStarted**: Server-started pattern to look for in the debug server output.
+
+**serverLaunchTimeout**: Time in milliseconds, for the debugger to wait for the debugServer to start up. Default is 10000.
+
+## Additional properties
+
+**processId**: Defaults to `${command.pickProcess}` which will display a list of available processes the debugger can attach to.  It is recommended to leave this default, but the property can be explicitly set to a specific process ID for the debugger to attach to.
+
+**request**: Indicates whether the configuration section is intended to `launch` the program or `attach` to an already running instance.
+
+**targetArchitecture**: `Deprecated` This option is no longer needed as the target architecture is automatically detected.
```

Comment:
@jacdavis, was this correctly fixed? I know that we had to remove some of Paul's code that did the architecture detection because it was blocking multi-threaded apps.